napari/_qt/containers/_base_item_model.py

```diff
@@ -14,6 +14,9 @@
 
 ItemType = TypeVar("ItemType")
 
+ItemRole = Qt.UserRole
+SortRole = Qt.UserRole + 1
```

Comment:
Not something that I want to hold up this PR (especially since it's not *really* new code), but my English -> unfamiliar code parser is breaking here. Can you explain what these mean? Possibly a link to relevant docs? My quick Google search only took me to confused users and unhelpful responses. :joy:

Reply:
See https://doc.qt.io/qt-5/qt.html#ItemDataRole-enum

Reply:
That definitely clarifies one half of it, thanks @goanpeca! But I don't understand the very concept of "roles"...

Reply:
I can add an extended docstring somewhere to explain it... since it's kind of important for anyone who would want to subclass this for something other than the layer list.  but the main idea is that when you try to get information about an item in the model, rather than just getting the item itself (like `item = model[index]`), you ask for certain "roles" like `text = model.data(index, Qt.DisplayRole)` or `icon = model.data(index, Qt.DecorationRole)` or `visible = model.data(index, Qt.CheckedRole)`, etc... [Here's the key part with the layerlist](https://github.com/tlambert03/napari/blob/ac6427fdfae4bc29b162c51901d76ab0b86ccfd0/napari/_qt/containers/qt_layer_model.py#L11-L38).  Most of those roles have builtin-meaning: like Qt builds the bulk of the actual visual appearance of each "widget" in the layer list based off of those little "roles" of data (though they're not actually QWidgets in this case)... but you can also create your own roles in certain cases.
in summary: the `model.data()` method is the heart of the translation between the object that is actually _in_ the model, and anything else (like a view) that wants to present some information about the items in a model